### DISCUSSION TITLE
A suggestion to add an exception to help troubleshoot MongoDB projections.

### DISCUSSION THREAD

**Discussion by ru-ng** [Eventuous/eventuous#286] [General]
I noticed that when events had the same name in the MapEvents() function, events just stops projecting to MongoDB. I spent quite a long time trying to pinpoint the problem and finally realised that our integration event name was the same as our domain event name and Eventuous wasn't sure which one it was referring to. I understand that it was our fault but it would be nice if Eventuous stopped with an exception instead of quietly running on. After renaming the integration event name, the MongoDB projections started to work.

**Comment by alexeyzimarev** [comment]
If you enable Eventuous logs, you will see log messages from different Eventuous components, including the type mapper. It logs a warning when event names are conflicting https://github.com/Eventuous/eventuous/blob/a66fb591bb82c68fcba172a3f4ef26f073dedff0/src/Extensions/src/Eventuous.Extensions.DependencyInjection/README.md?plain=1#L13
